src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.xaml.cs

```diff
@@ -0,0 +1,777 @@
+		}
+
+		if (_scrollH is not null)
+		{
+			_scrollH.Scroll += ScrollH_Scroll;
+		}
+	}
+
+	private double _previousVerticalScrollValue = double.MinValue;
+
+	private double _previousHorizontalScrollValue = double.MinValue;
+
+	//TemplateBinding doesn't support TwoWay mode. We need to manually update the values
+	private void ScrollV_Scroll(object sender, ScrollEventArgs e)
+	{
+		if ((_previousVerticalScrollValue > e.NewValue && !CanScrollUp()) ||
+			(_previousVerticalScrollValue < e.NewValue && !CanScrollDown()))
+		{
+			return;
+		}
+
+		VerticalOffset = -1 * e.NewValue;
+		_previousVerticalScrollValue = e.NewValue;
+	}
+
+	private void ScrollH_Scroll(object sender, ScrollEventArgs e)
+	{
+		if ((_previousHorizontalScrollValue < e.NewValue && !CanScrollRight()) ||
+			(_previousHorizontalScrollValue > e.NewValue && !CanScrollLeft()))
+		{
+			return;
+		}
+
+		HorizontalOffset = -1 * e.NewValue;
+		_previousHorizontalScrollValue = e.NewValue;
+	}
+
+	#endregion
+
+	private uint _capturedPointerId;
+	private Point _referencePosition;
+
+	private void RegisterPointerHandlers()
+	{
+		// Register for pointer events.
+		PointerPressed -= OnPointerPressed;
+		PointerReleased -= OnPointerReleased;
+		PointerMoved -= OnPointerMoved;
+		PointerWheelChanged -= OnPointerWheelChanged;
+
+		PointerPressed += OnPointerPressed;
+		PointerReleased += OnPointerReleased;
+		PointerMoved += OnPointerMoved;
+		PointerWheelChanged += OnPointerWheelChanged;
+	}
+
+	private void OnPointerPressed(object sender, PointerRoutedEventArgs e)
+	{
+		if (!IsAllowedToWork)
+		{
+			return; // Don't handle the event when the control is disabled.
+		}
+
+		var pointerPoint = e.GetCurrentPoint(this);
+		var pointerProperties = pointerPoint.Properties;
+
+		// If the middle button of a mouse is pressed, then we want to handle the event.
+		if (pointerProperties.IsMiddleButtonPressed
+#if IS_WINUI
+			&& pointerPoint.PointerDeviceType == PointerDeviceType.Mouse)
+#else
+			&& pointerPoint.PointerDevice.PointerDeviceType == PointerDeviceType.Mouse)
+#endif
+		{
+			e.Handled = true;
+
+			// Capture the pointer so that we can track its movement even if it leaves the bounds of the control.
+			var pointer = e.Pointer;
+			var captured = CapturePointer(pointer);
+			if (captured)
+			{
+				_capturedPointerId = pointer.PointerId;
+				_referencePosition = pointerPoint.Position;
+				_lastPosition = _referencePosition;
+			}
+		}
+	}
+
+	private void OnPointerReleased(object sender, PointerRoutedEventArgs e)
+	{
+		ReleasePointerCaptures();
+		_capturedPointerId = default;
+	}
+
+	private void OnPointerMoved(object sender, PointerRoutedEventArgs e)
+	{
+		if (!(IsAllowedToWork && _capturedPointerId > 0 && IsPanAllowed))
+		{
+			return;
+		}
+
+		var currentPosition = e.GetCurrentPoint(this).Position;
+		_movimentDirection = (currentPosition.X > _lastPosition.X, currentPosition.Y > _lastPosition.Y);
+		_lastPosition = currentPosition;
+
+		if (CanMoveThisWay())
```

Comment:
naming of this method, raises questions..?
maybe consider passing _mov~~i~~ementDirection as parameter:
```suggestion
		if (CanMoveIn(_movimentDirection))
```